src/app/modules/group/components/group-log-view/group-log-view.component.ts

```diff
@@ -88,4 +115,42 @@ export class GroupLogViewComponent implements OnChanges, OnDestroy {
     }));
   }
 
+  onMouseEnter(event: Event, itemId: string, index: number): void {
+    const targetRef = this.contentRef?.get(index);
+    if (!targetRef) {
+      throw new Error('Unexpected: Target is not found');
+    }
+    this.showOverlaySubject$.next({ event, itemId, target: targetRef.nativeElement });
+  }
+
+  onMouseLeave(event: MouseEvent, field: string): void {
+    if (field !== 'item.string.title') {
+      return;
+    }
+
+    const target = event.target;
+    const relatedTarget = event.relatedTarget;
+    let closeOverlay = true;
+
+    if (target instanceof HTMLElement && relatedTarget instanceof HTMLElement) {
+      let currentElement = relatedTarget;
+      while (currentElement.parentElement) {
+        if (currentElement.classList.contains('alg-path-suggestion-overlay')) {
+          closeOverlay = false;
+          break;
+        }
+        currentElement = currentElement.parentElement;
+      }
+    }
+
+    if (closeOverlay) {
+      this.closeOverlay();
+    }
```

Comment:
I think this code will become hard to maintain, there should be a more expressive/declarative way to write it.
If my understanding is correct, given that the overlay is a singleton (same one is reused for each cell), we have the following cases for a specific cell:
1. No overlay is opened **or** overlay is opened for another cell
   - mouse leaves cell: noop
   - mouse enters cell: trigger a "path suggestion" call and show overlay for current cell
2. An overlay is opened for current cell
   - mouse leaves current cell and does not enter overlay: close overlay
   - mouse leaves current cell and enters overlay: noop
   - mouse leaves overlay and enters current cell: noop

This is obviously non-trivial, but rx might help us here :thinking: 
Here's a draft that might help, what do you think?
```ts
closeOverlaySubscription?: Subscription // to be cleared on destroy as well

onMouseEnter(event, itemId, index) {
    // clear previous subscription if any
    this.closeOverlaySubscription?.unsubscribe();

  // current treatment. fetch etc.
  // and at the end: (this code is designed to entirely replace the `onMouseLeave()` method)

  // add a requestAnimationFrame to give time to the overlay to get rendered
  // Might not be needed, to be checked!
  requestAnimationFrame(() => {
    this.closeOverlaySubscription = merge(
      // mouseleave event is triggered before mouseenter of another target
      fromEvent(this.op, 'mouseleave').pipe(map(() => true)), // close overlay on mouse leave overlay
      fromEvent(event.target, 'mouseleave').pipe(map(() => true)), // close overlay on mouse leave cell
      fromEvent(this.op, 'mouseenter').pipe(map(() => false)), // cancels close when mouse enters overlay
      fromEvent(event.target, 'mouseenter').pipe(map(() => false)), // cancels close when mouse (re)enters current cell
    ).pipe(
      // consider only latest value in a time window of 500ms so that:
      // if the user leaves the cell to enter the overlay within 500ms, the last `shouldClose` value will be `false`
      // if the user leaves the overlay to come back to the cell within 500ms, the last `shouldClose value will be `false`
      debounceTime(500),
      filter(shouldClose => shouldClose === true),
    ).subscribe(() => this.closeOverlay())
  })
}
```

Reply:
I've tried to use this way, but after spend a lot of time still can't get stabile work, it makes a lot additional problems. Actually I've spend a lot of time to get stable work for current implementation, so really not sure do we need it right now, otherwise feature will became too expensive.

Reply:
I agree, intensive usage of rxjs will not make it easier in the future, and it will probably make it even harder.

The part I found hard to maintain or to clearly see what it does is the `while`+`if` loop, and to me it can be replaced by the `Node.closest(selector)` method, see #1030 proposal.